### ISSUE TITLE
PowerShell session should not "remember" debug commands in history

### ISSUE THREAD

**Issue by spencerboatepic** [PowerShell/PowerShell#24354]
### Summary of the new feature / enhancement

When stepping through a function, one usually "spams" a combination of 'v' 'n' 'c' in debug mode. Once the function ends, or the user enters 'q', debug mode ends but those collection of debug commands are saved in the history of inputted commands, making it harder to find relevant commands in the history (pressing up on the keyboard) while searching through it. These debug commands should be disposed of from the history when debug mode ends because they are no longer useful in the scope of the session.

**Comment by mklement0** [comment]
Good point, but note that this problem must be addressed in [PSReadLine](https://github.com/PowerShell/PSReadLine).

Perhaps a team member can transfer your issue there.

**Comment by kilasuit** [comment]
I was looking into this briefly and this would need to be added in both PSReadLine and also in core product due to duplications in how history items are dealt with so would need an issue in both to track and manage and update the areas where this is needed.

I'll get an issue in PSReadLine raised but it won't be a quick fix in either & would be an opt-out mechanism if implemented

**Comment by kilasuit** [comment]
[Issue raised in PSReadLine](https://github.com/PowerShell/PSReadLine/issues/4198)

**Comment by mklement0** [comment]
Thanks, @kilasuit, but I'm not sure that there's anything to be done in _this_ repo:

* `Get-History` does _not_ show the debugging commands.

* While interactive command recall with up-arrow _does_, I think this is a feature of `conhost.exe` and Windows Terminal windows over which PowerShell presumably has no control.

**Comment by kilasuit** [comment]
Ahh yeah I forgot that the up history of conhost/terminal is a thing too and likely we can't do anything about that in PowerShell, thou keeping this open for the WG to discuss

**Comment by iRon7** [comment]
Related: #20738 *`Write-Debug` expands `-Message` when it should `SilentlyContinue`*

**Comment by kborowinski** [comment]
Thankfully there is an `$PSDebugContext` automatic variable that is available **while** debugging, so you can add this little snippet to your profile to make sure that debug commands are not saved:
```powershell
Set-PSReadLineOption -AddToHistoryHandler {-not (Test-Path -Path 'Variable:PSDebugContext')}
```

**Comment by theJasonHelmick** [comment]
Thank you @spencerboatepic for this issue. The Interactive WG has reviewed this and recommends the solution described by @kborowinski.  This give folks the option on whether they want or do not want the commands in history.

**Comment by SeeminglyScience** [comment]
One extra tip, you don't actually have to spam them. Pressing enter at a blank prompt while debugging will repeat the previous command.
